### ISSUE TITLE
Add a11y functionality to drag-drop

### ISSUE THREAD

**Issue by drstonebraker** [angular/components#18291]
#### Feature Description

drag-drop should have built in a11y functionality for keyboard-only use, or an api for enabling such.

#### Use Case

keyboard-only use

**Comment by literalpie** [comment]
Also, some sort of screenreader consideration would be valuable
